test/widgets/dialog_checks.dart

```diff
@@ -12,15 +14,32 @@ Widget checkErrorDialog(WidgetTester tester, {
   required String expectedTitle,
   String? expectedMessage,
 }) {
-  final dialog = tester.widget<AlertDialog>(find.byType(AlertDialog));
-  tester.widget(find.descendant(matchRoot: true,
-    of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
-  if (expectedMessage != null) {
-    tester.widget(find.descendant(matchRoot: true,
-      of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
+  switch (defaultTargetPlatform) {
+    case TargetPlatform.android:
+    case TargetPlatform.fuchsia:
+    case TargetPlatform.linux:
+    case TargetPlatform.windows: {
+      final dialog = tester.widget<Dialog>(find.byType(Dialog));
+      tester.widget(find.widgetWithText(Dialog, expectedTitle));
+      if (expectedMessage != null) {
+        tester.widget(find.widgetWithText(Dialog, expectedMessage));
```

Comment:
This test logic in the non-Cupertino case differs from the old test logic. But the behavior in the non-Cupertino case is supposed to be the same as it was before, right? What's the motivation for changing the test logic?

Reply:
Ah I see, the old version fails because there's no longer a widget whose exact type is `AlertDialog` — the widget is an instance of a subclass instead — and `find.byType` searches by exact type.

A clean solution to that is to search by type in a way that includes subclasses. With the original tests plus just this tweak, all the tests pass:
```diff
-  final dialog = tester.widget<AlertDialog>(find.byType(AlertDialog));
+  final dialog = tester.widget<AlertDialog>(find.bySubtype<AlertDialog>());
```

I found `bySubtype` by looking at the doc for `byType`, and in particular the "See also" section at the end:
```dart
  /// See also:
  /// * [bySubtype], which allows subtype tests.
```